src/others/cpp_tips.md

```diff
@@ -0,0 +1,27 @@
+---
+tags:
+  - Original
+---
+
+# C++ Tips and Tricks for Competition Programming
+
+## Faster I/O
+
+- `std::ios_base::sync_with_stdio(false)`: by default, C++ works with C++ iostreams (like `std::cin`) and C style stdio streams (like `stdin`) by synchronizing them.
+Since you should never need C style I/O (like `scanf`, `printf`), in C++, disabling this synchronization will make C++ I/O competitive with C style `scanf`.
+- `std::cin.tie(0)`: By default, iostreams flush cout every time you read from cin, so that you will always display any output before reading any input.
+If you are not solving an interactive problem, i.e. you don't require interleaving inputs and outputs, you can disable this automatic flushing to make I/O faster.
```

Comment:
For interactive problems, you can as well use `cout.flush()` or `cout << endl;` to flush output manually, rather than disabling `cin.tie(0)`.

Reply:
Noted.